packages/blade/src/components/Tour/Tour.web.tsx

```diff
@@ -0,0 +1,199 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+/* eslint-disable consistent-return */
+/* eslint-disable @typescript-eslint/explicit-function-return-type */
+import React, { useState, useCallback, useMemo } from 'react';
+import { FloatingPortal } from '@floating-ui/react';
+import { TourContext } from './TourContext';
+import { TourPopover } from './TourPopover';
+import {
+  smoothScroll,
+  useDelayedState,
+  useIntersectionObserver,
+  useIsTransitioningBetweenSteps,
+  useLockBodyScroll,
+} from './utils';
+import type { TourMaskRect, TourProps } from './types';
+import { TourMask } from './TourMask';
+import { transitionDelay } from './tourTokens';
+import { useTheme } from '~utils';
+import { useIsomorphicLayoutEffect } from '~utils/useIsomorphicLayoutEffect';
+
+const Tour = ({
+  steps,
+  activeStep,
+  isOpen,
+  onFinish,
+  onOpenChange,
+  onStepChange,
+  children,
+}: TourProps): React.ReactElement => {
+  const { theme } = useTheme();
+  const [refIdMap, setRefIdMap] = useState(new Map<string, React.RefObject<HTMLElement>>());
+  const [size, setSize] = useState<TourMaskRect>({
+    x: 0,
+    y: 0,
+    height: 0,
+    width: 0,
+  });
+
+  // delayed state is used to let the transition finish before reacting to the state changes
```

Comment:
Which transition needs to be finished before we starT?

Reply:
The delay between going from 1 step to another. 

Check the motion on [figma](https://www.figma.com/file/LSG77hEeVYDk7j7WV7OMJE/Blade-DSL---Components-Guideline?type=design&node-id=4406-63721&mode=design&t=cG0DF5B4SwLIVA4i-0) to see what I mean: 
<img width="887" alt="image" src="https://github.com/razorpay/blade/assets/35374649/01371c25-a14f-4c40-b79a-975078a96d13">